theano/gof/opt.py

```diff
@@ -3039,3 +3039,31 @@ def check_stack_trace(f_or_fgraph, ops_to_check='last', bug_print='raise'):
                 return False
 
     return True
+
+
+class CheckStrackTraceFeature(object):
+    def on_import(self, fgraph, node, reason):
+        if theano.config.check_stack_trace == 'check_all':
+            if not check_stack_trace(fgraph, 'all'):
+                raise NotImplementedError(
+                    "Empty stack trace! The optimization that whose"
+                    " stacktrace is empty is" + str(reason))
+        if theano.config.check_stack_trace == 'check_and_skip':
+            if not check_stack_trace(fgraph, 'all'):
+                apply_nodes_to_check = fgraph.apply_nodes
+                for node in apply_nodes_to_check:
+                    for output in node.outputs:
+                        if not hasattr(output.tag, 'trace') or not output.tag.trace:
+                            output.tag.trace = [[('/dummy/path', 0, 'This is a dummy value! The stack trace of ' +
+                                                 str(reason) + ' was empty!', 'dummy(value)')]]
```

Comment:
Actually, I would re-use the same message as above (it is the stack trace of the creation of the variable, not of the optimization).
Is it possible to have empty strings for the dummy values?